apps/api/src/domain/models/context/repositories/arango-note-query-repository.spec.ts

```diff
@@ -0,0 +1,321 @@
+
+            expect(role).toBe(translationRole);
+        });
+    });
+
+    describe(`addAudio`, () => {
+        describe(`when there is no audio to begin with`, () => {
+            const audioId = buildDummyUuid(44);
+
+            const existingAudio = buildTestInstance(EventSourcedAudioItemViewModel, {
+                id: audioId,
+            });
+
+            const originalLanguageCode = LanguageCode.English;
+
+            const noteWithNoAudio = buildTestInstance(EventSourcedNoteViewModel, {
+                id: buildDummyUuid(1),
+                note: buildMultilingualTextWithSingleItem(
+                    'existing text for target language',
+                    originalLanguageCode
+                ),
+            });
+
+            beforeEach(async () => {
+                await databaseProvider.clearViews();
+
+                await testQueryRepository.create(noteWithNoAudio);
+
+                // @ts-expect-error FIX ME
+                await new ArangoAudioItemQueryRepository(connectionProvider).create(existingAudio);
+            });
+
+            it(`should have a test`, async () => {
+                await testQueryRepository.addAudio(
+                    noteWithNoAudio.id,
+                    audioId,
+                    originalLanguageCode
+                );
+
+                const updatedView = (await testQueryRepository.fetchById(
+                    noteWithNoAudio.id
+                )) as EventSourcedNoteViewModel;
+
+                expect(updatedView.audio.hasAudioIn(originalLanguageCode)).toBe(true);
+
+                expect(updatedView.audio.getIdForAudioIn(originalLanguageCode)).toBe(audioId);
+            });
+        });
+
+        describe(`when there is audio for the original language, but the audio is being added for the translation language`, () => {
+            it.todo(`should have a test`);
+        });
+    });
+
+    describe(`createNoteAbout`, () => {
+        const resourceCompositeIdentifier = {
+            type: WIDGET_TYPE,
+            id: buildDummyUuid(123),
+        };
+
+        const testWidget = new WidgetViewModel({
+            ...resourceCompositeIdentifier,
+            rating: 100,
+        });
+
+        const note = buildTestInstance(EventSourcedNoteViewModel, {
+            id: buildDummyUuid(5),
+            connectedResources: {
+                self: {
+                    ...testWidget,
+                    context: generalContext,
+                },
+            },
+        });
+
+        beforeEach(async () => {
+            await databaseProvider.clearViews();
+
+            await databaseProvider.getDatabaseForCollection('widget__VIEWS').clear();
+
+            await databaseProvider
+                .getDatabaseForCollection('widget__VIEWS')
+                .create(mapEntityDTOToDatabaseDocument(testWidget));
+        });
+
+        it(`should create the note`, async () => {
+            // TODO we don't want the full note here
```

Comment:
We need to update the interface so that only composite IDs, but not full view models are required on the `createNoteAbout` and `connectResources` methods. It is the repositories job to join in a denormalized view of the resource.